include/zephyr/drivers/bluetooth/hci_driver.h

```diff
@@ -21,7 +21,10 @@
 #include <zephyr/net/buf.h>
 #include <zephyr/bluetooth/buf.h>
 #include <zephyr/bluetooth/hci_vs.h>
-#include <zephyr/device.h>
+
+
+/* Opaque type representing a Zephyr device. */
+struct device;
```

Comment:
This seems like a pretty bad solution IMO; we are basically defining an opaque struct in `hci_driver.h` (a public header) in the `bluetooth` directory. 

If there an issue with including `device.h` it sounds like `device.h` should be modified, and not this public header file.

Reply:
we're doing the same thing in bluetooth.h, take a look

Reply:
Can you point me to the place in `bluetooth.h` where we are doing the same thing? I don't see anything where a Bluetooth header is defining a struct outside the `bt_` namespace. 

I do, however, see multiple other public header files do this exact thing, even if it is outside their namespaces. 

@jori-nordic I don't understand the issue that this PR is supposed to fix, can you elaborate? Maybe that will help me to get a more positive view on this :)

Reply:
You're right, bluetooth.h is not defining a struct outside the bt_ namespace.

The issue, in this case, is that including `device.h` drags in a whole lot of DT stuff that we don't want/need in the unit-tests.

The only thing we need in `device.h` is the `struct device` type, which for all intents and purposes is just a void ptr passed thru `bt_hci_transport_setup`. 
What I'm trying to do with this PR is express this in code I guess.

I thought this was acceptable, since as you said, other headers are already doing it (e.g. `pm/device.h`, `gsm_ppp.h`).
And I think `hci_driver.h` isn't a public API header, it's only used internally and for hci drivers.

Reply:
> And I think hci_driver.h isn't a public API header, it's only used internally and for hci drivers.

It is placed in the `include/zephyr` directory, even though it is almost exclusively used internally. 

> The issue, in this case, is that including device.h drags in a whole lot of DT stuff that we don't want/need in the unit-tests.

So this is a unit-test issue only? It kind of makes sense that we are not using `device.h` in the unit tests, as they are unlikely to be testing anything for any specific boards. Would it be possible to supply some sort of "dummy" or "unittest-only" `device.h` header when building for unittests instead? At least to me, that seems like a better long-term solution, unless it will require significant effort.

Reply:
I'm a bit against a dummy `device.h`. It's another thing to maintain and keep in sync with the original. It would be ok if the device tree subsystem provided the dummy `device.h`. Or maybe there could be a `device_minimal.h` that basically is just the forward declarations?

But I wonder if the correct way is to have an "empty"/default board for unit testing?

Reply:
I agree with you @alwa-nordic . Ideally for (most) UTs, we'd like to not have any DT parsing or macros.
I'd be okay with `device_minimal.h` (or `device_def.h`) with just the forward declarations, and include that in `device.h` if that also suits you @Thalley .

Reply:
It's better than defining a non-BT struct in `bluetooth.h` :D 

However, this still sounds like a unittest-only issue, so perhaps the forward declaration could also just be defined in a local header or source file for the unittests? i.e. add `struct device;` to whatever source or header file that needs it to be declared for that specific case.

Reply:
yeah the problem is not that the `device.h` header is not included, but rather that it is included.
so the stack source will need to be modified for either of those two solutions. 
Both are fine with me, even if I have a slight preference for alwa's.